receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go

```diff
@@ -76,18 +76,35 @@ var defaultCreateManager = func(memoryCache *memory.InMemoryCache, sysfs sysfs.S
 	return manager.New(memoryCache, sysfs, houskeepingConfig, includedMetricsSet, collectorHTTPClient, rawContainerCgroupPathPrefixWhiteList, perfEventsFile)
 }
 
-type cadvisorOption func(*Cadvisor)
+// Option is a function that can be used to configure Cadvisor struct
+type Option func(*Cadvisor)
 
-func cadvisorManagerCreator(f createCadvisorManager) cadvisorOption {
+func cadvisorManagerCreator(f createCadvisorManager) Option {
 	return func(c *Cadvisor) {
 		c.createCadvisorManager = f
 	}
 }
 
+// DecoratorOption constructs an option for configuring the metric decorator
+func DecoratorOption(d decorator) Option {
+	return func(c *Cadvisor) {
+		c.k8sDecorator = d
+	}
+}
+
 type hostInfo interface {
 	GetNumCores() int64
 	GetMemoryCapacity() int64
 	GetClusterName() string
+	GetEBSVolumeID(string) string
+	ExtractEbsIDsUsedByKubernetes() map[string]string
+	GetInstanceID() string
+	GetInstanceType() string
+	GetAutoScalingGroupName() string
+}
+
+type decorator interface {
```

Comment:
```suggestion
type Decorator interface {
```

Since this interface consists of exported functions and types, and it appears to be implemented outside of this package, it probably makes sense to export the interface as well to allow implementers to validate their implementation using this pattern:

```go
var _ cadvisor.Decorator = DecoratorImpl{}
```

Reply:
Good suggestion!